lib/iiif_manifest/v3/manifest_builder/thumbnail_builder.rb

```diff
@@ -0,0 +1,52 @@
+module IIIFManifest
+  module V3
+    class ManifestBuilder
+      class ThumbnailBuilder
+        attr_reader :display_content, :iiif_thumbnail_factory, :image_service_builder_factory
+        def initialize(display_content, iiif_thumbnail_factory:, image_service_builder_factory:)
+          @display_content = display_content
+          @iiif_thumbnail_factory = iiif_thumbnail_factory
+          @image_service_builder_factory = image_service_builder_factory
+        end
+
+        # @param canvas_or_manifest [#thumbnail=]
+        def apply(canvas_or_manifest)
+          build_thumbnail
+          image_service_builder.apply(thumbnail)
+          canvas_or_manifest.thumbnail = thumbnail
+        end
+
+        private
+
+        def build_thumbnail
+          thumbnail['id'] = File.join(display_content.iiif_endpoint.url, 'full', '!200,200', '0', 'default.jpg')
+          thumbnail['height'] = (display_content.height * reduction_ratio).round
+          thumbnail['width'] = (display_content.width * reduction_ratio).round
+          thumbnail['format'] = display_content.format
+        end
+
+        def reduction_ratio
+          width = display_content.width
+          height = display_content.height
+          max_edge = 200.0
+          return 1 if width <= max_edge && height <= max_edge
+
+          long_edge = [height, width].max
+          max_edge / long_edge
+        end
+
+        def thumbnail
+          @thumbnail ||= iiif_thumbnail_factory.new
+        end
+
+        def iiif_endpoint
+          display_content.try(:iiif_endpoint)
+        end
+
+        def image_service_builder
```

Comment:
Is there a problem if the image_service_builder_factory receives `nil`?

Reply:
looks like down the line, it will affect it here: https://github.com/samvera/iiif_manifest/blob/main/lib/iiif_manifest/v3/manifest_builder/image_service_builder.rb#L12-L14
are you thinking lonely operators for those three lines?